packages/rsocket-websocket-client/src/RSocketWebSocketClient.js

```diff
@@ -66,7 +68,17 @@ export default class RSocketWebSocketClient implements DuplexConnection {
         'established.',
     );
     this._setConnectionStatus(CONNECTION_STATUS.CONNECTING);
-    const socket = (this._socket = new WebSocket(this._options.url));
+
+    const wsCreator = this._options.wsCreator;
+    const url = this._options.url;
+    const wsOptions = this._options.wsOptions;
+    if (wsCreator) {
```

Comment:
I think ternary if is clearer here, but up to you
`this._socket = wsCreator ? wsCreator(url) : new WebSocket(url);